be/src/olap/delta_writer.cpp

```diff
@@ -66,7 +66,7 @@ DeltaWriter::DeltaWriter(WriteRequest* req, StorageEngine* storage_engine, Runti
                          const UniqueId& load_id)
         : _req(*req),
           _rowset_builder(*req, storage_engine, profile),
-          _memtable_writer(*req, profile),
+          _memtable_writer(new MemTableWriter(*req, profile)),
```

Comment:
这里有一个问题，就是这个profile，虽然memtable writer 是一个shared ptr，可以再 mmemtable_memory_limiter 中lock 一下，但是假如delta writer 推出了，那么profile 这个指针关联的对象可能也没有了；但是 memtablewriter 依赖了profile，那么在memtable memory limiter 那个线程中，就可能挂掉。
可能我们需要把load channel 这里的profile 改成shared ptr

Reply:
It's not easy to use shared_ptr for profile, because it's managed by internal object pool. All sub profile will share same life-cycle with the parent profile.

Here's a workaround, I have removed dependency on profile except in `close_wait()`, which should not be called in MemTableMemoryLimiter anyways. 9a9b368eb65575ebc497c7c18f9fd26e94ddd522

Reply:
Updated in ef18a5efaa3b30ae29c7ad4cb8892edd19cb0daf : MemTableWriter object no longer own member `_profile`